test/serializers/associations_test.rb

```diff
@@ -145,6 +145,26 @@ def test_associations_custom_keys
         assert expected_association_keys.include? :writer
         assert expected_association_keys.include? :site
       end
+
+      def test_belongs_to_doesnt_load_record
+        post = Post.new({ id: 997 })
+
+        class << post
+          def blog_id
+            996
+          end
+          def blog
+            fail "should use blog_id"
+          end
+        end
+
+        post_serializer = Class.new(ActiveModel::Serializer) do
```

Comment:
Agreed

Le lundi 31 août 2015, ssendev <notifications@github.com> a écrit :

> In test/serializers/associations_test.rb
> <https://github.com/rails-api/active_model_serializers/pull/1100#discussion_r38342653>
> :
>
> > @@ -145,6 +145,26 @@ def test_associations_custom_keys
> >          assert expected_association_keys.include? :writer
> >          assert expected_association_keys.include? :site
> >        end
> > +
> > +      def test_belongs_to_doesnt_load_record
> > +        post = Post.new({ id: 997 })
> > +
> > +        class << post
> > +          def blog_id
> > +            996
> > +          end
> > +          def blog
> > +            fail "should use blog_id"
> > +          end
> > +        end
> > +
> > +        post_serializer = Class.new(ActiveModel::Serializer) do
>
> very interesting. but in the referenced test it's required since that's
> what's being tested. here it seems like it would just make the test more
> complex for "no" benefit. especially considering there already is a
> PostSerializer which would have to be restored.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails-api/active_model_serializers/pull/1100/files#r38342653>
> .
>


-- 
Lucas Hosseini
lucas.hosseini@gmail.com